ksqldb-parser/src/main/java/io/confluent/ksql/parser/json/WindowExpressionDeserializer.java

```diff
@@ -20,13 +20,26 @@
 import com.fasterxml.jackson.databind.JsonDeserializer;
 import io.confluent.ksql.execution.windows.KsqlWindowExpression;
 import io.confluent.ksql.parser.ExpressionParser;
+import io.confluent.ksql.parser.OutputRefinement;
 import java.io.IOException;
 
 class WindowExpressionDeserializer<T extends KsqlWindowExpression> extends JsonDeserializer<T> {
   @Override
   @SuppressWarnings("unchecked")
   public T deserialize(final JsonParser parser, final DeserializationContext ctx)
       throws IOException {
-    return (T) ExpressionParser.parseWindowExpression(parser.readValueAs(String.class));
+    final String extendedWindowExpression = parser.readValueAs(String.class);
+
+    final KsqlWindowExpression windowExpression;
+
+    if (extendedWindowExpression.endsWith("FINAL")) {
```

Comment:
Instead of this hack, can we update the serializer/deserializer to serialize an additional, optional field to specify the emitStrategy? Ideally there'd be a way to deserialize a window object that is either represented as a raw string (legacy) or as an object (new).